### PR TITLE
Add xpath support

### PR DESCRIPTION
**Pull request by gwintrob**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -2,18 +2,34 @@ module.exports = {
...
+            }
+
+            return this.toElementArray(nodes);
         } else if(elements instanceof HTMLElement) {
             return [elements];
         } else {
-            return ((elements !== null) && elements.length)
-                ? Array.prototype.slice.call(elements)
-                : [];
+            if (elements === null || !elements.length) {
```
Comment (evanhl) [inline index.js]: This is much more readable now.

File: index.js [hunk 2]
```diff
@@ -2,18 +2,34 @@ module.exports = {
...
      */
-    'toElementArray' : function(elements) {
+    'toElementArray' : function(elements, isXpath) {
         if(typeof elements === 'string') {
-            return this.toElementArray(document.querySelectorAll(elements));
+            if (!isXpath) {
+                return this.toElementArray(document.querySelectorAll(elements));
+            }
+
+            var nodes = [],
```
Comment (evanhl) [inline index.js]: It would be nice to have the xpath to HTMLElement array conversion in its own method.

File: index.js [hunk 3]
```diff
@@ -2,18 +2,34 @@ module.exports = {
...
+                iterator = document.evaluate(elements, document.body),
+                node = iterator.iterateNext();
+
+            while (node) {
+                nodes.push(node);
+                node = iterator.iterateNext();
+            }
+
+            return this.toElementArray(nodes);
```
Comment (evanhl) [inline index.js]: Why is this call to `toElementArray` necessary?  Won't that just clone the array?

Comment (gwintrob) [inline index.js]: Done.

File: index.js (hunk 1 above)
Comment (gwintrob) [inline index.js]: Thanks!

File: index.js (hunk 2 above)
Comment (gwintrob) [inline index.js]: Done!

**Comment by evanhl** [comment]
Looks good.

**Pull request merged**